src/Analyzers/CSharp/Analyzers/UseSystemThreadingLock/CSharpUseSystemThreadingLockDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,268 @@
+        //
+        // Note: both this dictionary is written to concurrently in the analysis pass below.  This is safe as we never
+        // are actually mutating the dictionary itself (we're not adding/removing items).  We're just getting a
+        // reference to its tuple value and overwriting a bool in place within that tuple with the new value.  And we
+        // only move hte value in one direction.  For example 'canUse' only moved from 'true' to 'false' and 'wasLocked'
+        // only moves the value from 'false' to 'true'.
+        var potentialLockFields = new SegmentedDictionary<IFieldSymbol, (FieldDeclarationSyntax fieldDeclaration, CodeStyleOption2<bool> option, bool canUse, bool wasLocked)>();
+
+        foreach (var (field, fieldDeclaration, option) in fieldsArray)
+            potentialLockFields[field] = (fieldDeclaration, option, canUse: true, wasLocked: false);
+
+        // Register a syntax node callback to ensure the field's initializer is either missing, or is only instantiating
+        // the field with a new object.
+        context.RegisterSyntaxNodeAction(context =>
+        {
+            var cancellationToken = context.CancellationToken;
+            var fieldDeclaration = (FieldDeclarationSyntax)context.Node;
+
+            foreach (var (currentField, currentFieldDeclaration, _) in fieldsArray)
+            {
+                if (currentFieldDeclaration == fieldDeclaration)
+                {
+                    // If we have an initializer at the declaration site, it needs to be initialized with either `new
+                    // object()` or `new()`.  If not, we can't convert this.  Note: we confirmed this field only has a
+                    // single declarator when doing our initial pass.  So .Single is safe here.
+                    var variableDeclarator = currentFieldDeclaration.Declaration.Variables.Single();
+                    if (variableDeclarator.Initializer != null &&
+                        !IsSystemObjectCreationExpression(context.SemanticModel, variableDeclarator.Initializer.Value, cancellationToken))
+                    {
+                        GetValueRefOrNullRef(potentialLockFields, currentField).canUse = false;
+                    }
+                }
+            }
+        }, SyntaxKind.FieldDeclaration);
+
+        // Now go see how the code within this named type actually uses any fields within.
+        context.RegisterOperationAction(context =>
+        {
+            var cancellationToken = context.CancellationToken;
+            var fieldReferenceOperation = (IFieldReferenceOperation)context.Operation;
+            var fieldReference = fieldReferenceOperation.Field.OriginalDefinition;
+
+            // We only care about examining field references to the fields we're considering converting to System.Threading.Lock.
+            if (!potentialLockFields.ContainsKey(fieldReference))
+                return;
```

Comment:
💡 Consider using `GetValueRefOrNullRef` here, and checking it for:

1. A null ref
2. canUse already false

Reply:
yup.  done.